rep-0150.rst

```diff
@@ -0,0 +1,208 @@
+
+This document covers the common metapackages used for recommended
+installations.
+It is expressed in the ros/metapackages repository on github.com
+[4]_.
+
+This REP proposes the same entrypoints as REP 108 and merely updates
+the variant definitions to reflect the organizational changes in ROS
+stacks.
+In particular, in ROS Melodic the `robot` metapackage is contracted to remove
+`robot_model` and instead explicitly list the necessary packages.
+
+The definitions in this REP are more focused and concise for clearer
+purpose, and do not list all transitive dependencies explicitly.
+
+
+Specification
+=============
+
+End-user entry points
+---------------------
+
+We define the same three main entry points for ROS users as REP 108
+[1]_.
+
+ * desktop_full (recommended)
+ * desktop
+ * ros_base
+
+Metapackages
+------------
+
+ROS Core
+''''''''
+
+The `ros_core` metapackage composes the core communication protocols.
+It may not contain any GUI dependencies.
+
+::
+
+ - ros_core:
+      extends: [ros, ros_comm]
+      packages: [catkin, cmake_modules, common_msgs, console_bridge,
+                 gencpp, geneus, gennodejs, genlisp, genmsg, genpy,
+                 message_generation, message_runtime, rosbag_migration_rule,
+                 rosconsole_bridge, roscpp_core, rosgraph_msgs, roslisp,
+                 rospack, std_msgs, std_srvs]
+
+ROS Base
+''''''''
+
+The `ros_base` metapackage composes the `ros_core` metapackage with
+commonly used libraries.
+It may not contain any GUI dependencies.
+
+::
+
+  - ros_base:
+      extends: [ros_core]
+      packages: [actionlib, angles, bond_core, class_loader,
+                 dynamic_reconfigure, nodelet_core,
+                 pluginlib]
+
+Robot metapackage
+'''''''''''''''''
+
+The `robot` metapackage is defined to be core, stable, ROS libraries
+for any robot hardware.
+It is the "general robotics" libraries of ROS.
+It may not contain any GUI dependencies.
+
+::
+
+  - robot:
+      extends: [ros_base]
+      packages: [collada_parser, control_msgs, diagnostics, executive_smach,
+                 filters, geometry, joint_state_publisher, kdl_parser,
+                 kdl_parser_py, robot_state_publisher, urdf,
+                 urdf_parser_plugin, xacro]
+
+Capability variants
+'''''''''''''''''''
+
+The capability variants organize commonly used libraries that are
+specific to a class of use case.
+We also define a `simulators` variant that provides an organizational
+role for higher-level variants.
+We discourage GUI dependencies in these stacks, if possible.
+
+::
+
+  - perception:
+      extends: [ros_base]
+      packages: [image_common, image_pipeline,
+                 image_transport_plugins, laser_pipeline,
+                 perception_pcl, vision_opencv]
+
+  - simulators:
+      extends: [robot]
+      packages: [gazebo_ros_pkgs, rqt_common_plugins, rqt_robot_plugins,
+                 stage_ros]
+
+  - viz:
+      extends: [robot]
```

Comment:
the metapackage `viz` [currently extends `ros_base`](https://github.com/ros/metapackages/blob/d56a5d422d02300ce6ce3cc50846053f9195bcd6/viz/package.xml#L13) and not `robot`

Reply:
Fixed.

Reply:
same here, not sure if the it's REP or the metapackage that should be modified. Maybe @tfoote or @dirk-thomas have some feedback?

Reply:
The robot target was to think what you'd want on the robot, whereas viz is for the developer interface so I think that it makes sense to stick with the ros_base.